From a4da077b8098e000b7791f1de6d59102c0f4ab68 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 15 Oct 2016 13:08:41 +0200
Subject: [PATCH] Remove dependency on GConf

It was used only in a11y applet to get /apps/metacity/general/visual_bell config. But metacity don't use this config a long time ago.
---
 configure.ac                                       |  2 +-
 debian/control                                     |  3 ---
 .../cinnamon/applets/a11y@cinnamon.org/applet.js   | 23 ----------------------
 3 files changed, 1 insertion(+), 27 deletions(-)

diff --git a/configure.ac b/configure.ac
index 147e108..db9e9d2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -75,7 +75,7 @@ PKG_CHECK_MODULES(CINNAMON, gio-2.0 >= $GIO_MIN_VERSION
                                gtk+-3.0 >= $GTK_MIN_VERSION
                                libmuffin >= $MUFFIN_MIN_VERSION
                                cjs-internals-1.0 >= $GJS_MIN_VERSION
-			       libcinnamon-menu-3.0 $recorder_modules gconf-2.0
+			       libcinnamon-menu-3.0 $recorder_modules
                                gdk-x11-3.0 libsoup-2.4 gl
 			       clutter-x11-1.0 >= $CLUTTER_MIN_VERSION
                                libstartup-notification-1.0 >= $STARTUP_NOTIFICATION_MIN_VERSION
diff --git a/debian/control b/debian/control
index 71df6c1..a4f0761 100644
--- a/debian/control
+++ b/debian/control
@@ -18,7 +18,6 @@ Build-Depends: debhelper (>= 9),
                libcaribou-dev,
                libcroco3-dev (>= 0.6.2),
                libdbus-glib-1-dev,
-               libgconf2-dev,
                libgirepository1.0-dev (>= 0.9.2),
                libglib2.0-dev (>= 2.37.3),
                libgnome-bluetooth-dev (>= 3.1.0),
@@ -68,7 +67,6 @@ Depends: cinnamon-common (= ${source:Version}),
          gnome-icon-theme-symbolic (>= 2.91),
          ${icon-theme:Depends},
          gir1.2-accountsservice-1.0,
-         gir1.2-gconf-2.0,
          gir1.2-gkbd-3.0,
          gir1.2-gnomebluetooth-1.0,
          gir1.2-keybinder-3.0,
@@ -91,7 +89,6 @@ Depends: cinnamon-common (= ${source:Version}),
          python3-dbus,
          python,
          python-dbus,
-         python-gconf,
          python-imaging,
          python-gi-cairo,
          python-gtk2,
diff --git a/files/usr/share/cinnamon/applets/a11y@cinnamon.org/applet.js b/files/usr/share/cinnamon/applets/a11y@cinnamon.org/applet.js
index c92a73a..9392fbc 100644
--- a/files/usr/share/cinnamon/applets/a11y@cinnamon.org/applet.js
+++ b/files/usr/share/cinnamon/applets/a11y@cinnamon.org/applet.js
@@ -2,7 +2,6 @@ const PopupMenu = imports.ui.popupMenu;
 const Gio = imports.gi.Gio;
 const Lang = imports.lang;
 const Applet = imports.ui.applet;
-const GConf = imports.gi.GConf;
 const Main = imports.ui.main;
 const Gdk = imports.gi.Gdk;
 
@@ -21,9 +20,6 @@ const DPI_FACTOR_LARGE   = 1.25;
 const DPI_FACTOR_LARGER  = 1.5;
 const DPI_FACTOR_LARGEST = 2.0;
 
-const KEY_META_DIR       = '/apps/metacity/general';
-const KEY_VISUAL_BELL = KEY_META_DIR + '/visual_bell';
-
 const DESKTOP_INTERFACE_SCHEMA = 'org.cinnamon.desktop.interface';
 const KEY_GTK_THEME      = 'gtk-theme';
 const KEY_ICON_THEME     = 'icon-theme';
@@ -54,9 +50,6 @@ MyApplet.prototype = {
             this.menu = new Applet.AppletPopupMenu(this, orientation);
             this.menuManager.addMenu(this.menu);            
                                 
-            let client = GConf.Client.get_default();
-            client.add_dir(KEY_META_DIR, GConf.ClientPreloadType.PRELOAD_ONELEVEL, null);
-
             let highContrast = this._buildHCItem();
             this.menu.addMenuItem(highContrast);
 
@@ -75,9 +68,6 @@ MyApplet.prototype = {
                                                                        'screen-keyboard-enabled');
             this.menu.addMenuItem(screenKeyboard);
 
-            let visualBell = this._buildItemGConf(_("Visual Alerts"), client, KEY_VISUAL_BELL);
-            this.menu.addMenuItem(visualBell);
-
             let stickyKeys = this._buildItem(_("Sticky Keys"), A11Y_SCHEMA, KEY_STICKY_KEYS_ENABLED);
             this.menu.addMenuItem(stickyKeys);
 
@@ -152,19 +142,6 @@ MyApplet.prototype = {
         return widget;
     },
 
-    _buildItemGConf: function(string, client, key) {
-        function on_get() {
-            return client.get_bool(key);
-        }
-        let widget = this._buildItemExtended(string,
-            client.get_bool(key),
-            client.key_is_writable(key),
-            function(enabled) {
-                client.set_bool(key, enabled);
-            });
-        return widget;
-    },
-
     _buildItem: function(string, schema, key) {
         let settings = new Gio.Settings({ schema_id: schema });
         let widget = this._buildItemExtended(string,
-- 
2.10.0

